### PR TITLE
fix(android): harden gradle injection edge cases

### PR DESCRIPTION
**Pull request by BlackishGreen33**
## Summary
- cover host `manifestPlaceholders` and missing/non-literal `versionName` with compileMods regression tests
- switch Android app/build.gradle injection to append JPush placeholders without relying on a `versionName` anchor

## Testing
- npm run build
- npm test -- --runInBand
- npm run lint

### CODE REVIEW

File: plugin/src/android/appBuildGradle.ts [hunk 1]
```diff
@@ -73,20 +151,36 @@ const getManifestPlaceholders = (
...
+
+  return [
+    `${indent}manifestPlaceholders += [`,
+    placeholders
+      .map((placeholder, index) => {
+        const separator = index === placeholders.length - 1 ? '' : ',';
+        return `${valueIndent}${placeholder}${separator}`;
+      })
+      .join('\n'),
+    `${indent}]`,
```
[the remark this answers is not in the document]
Comment (BlackishGreen33) [reply plugin/src/android/appBuildGradle.ts:173]: This precedence is intentional: the generated JPush placeholders are appended after the host map so unrelated host placeholders stay intact, while overlapping `JPUSH_*` keys end up using the plugin/build-time values. I kept that behavior and made the intent explicit inline in `appBuildGradle.ts` to avoid the parameter-order / merge-semantics ambiguity.
